### PR TITLE
Handle spread props in jsx-no-target-blank rule

### PR DESCRIPTION
**Pull request by randycoulman**
When running the `jsx-no-target-blank` rule against code like the following:

```
<a {...this.props} target='_blank' />
```

the following error would be raised:

```
TypeError: Cannot read property 'name' of undefined
```

The root cause is that the node representing the spread props does not have a `name` property.

Note that if the desired `rel` attribute appears before the spread props, the error does not occur.

### CODE REVIEW

File: tests/lib/rules/jsx-no-target-blank.js [hunk 1]
```diff
@@ -27,7 +27,8 @@ ruleTester.run('jsx-no-target-blank', rule, {
   valid: [
     {code: '<a href="foobar"></a>', parserOptions: parserOptions},
     {code: '<a randomTag></a>', parserOptions: parserOptions},
-    {code: '<a href="foobar" target="_blank" rel="noopener noreferrer"></a>', parserOptions: parserOptions}
+    {code: '<a href="foobar" target="_blank" rel="noopener noreferrer"></a>', parserOptions: parserOptions},
+    {code: '<a target="_blank" {...spreadProps} rel="noopener noreferrer"></a>', parserOptions: parserOptions}
```
Comment (ljharb) [inline tests/lib/rules/jsx-no-target-blank.js]: since the ordering of "rel" and the spread props seems to matter, should we also have a test for one where "rel" appears first?

File: lib/rules/jsx-no-target-blank.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = function(context) {
         var relFound = false;
         var attrs = node.parent.attributes;
         for (var idx in attrs) {
```
Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: @yannickcr is there any reason this uses `for..in` instead of `Object.keys(…).forEach(…)`?

File: tests/lib/rules/jsx-no-target-blank.js (hunk 1 above)
Comment (randycoulman) [inline tests/lib/rules/jsx-no-target-blank.js]: It only matters because we break out of the `for` loop as soon as we find the `rel` attribute we're looking for.  I could add another test case, but it would pass on the old code, so doesn't illustrate the problem.  I didn't think it worth it at the time, but if you think it is, I'm happy to add it.

Comment (ljharb) [inline tests/lib/rules/jsx-no-target-blank.js]: I think it's important to add, to prevent a regression - if there's already order-dependent bugs, there could easily be more added later.

Comment (randycoulman) [inline tests/lib/rules/jsx-no-target-blank.js]: Good thought.  Added in 3db97e49912ce1cccffbbebb595d9bfb14f65512.

File: lib/rules/jsx-no-target-blank.js (hunk 2 above)
Comment (yannickcr) [inline lib/rules/jsx-no-target-blank.js]: @ljharb If I remember correctly `Object.keys` is only supported since Node.js 4.0

Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: @yannickcr `Object.keys` is ES5, and is supported since node 0.6.

Comment (yannickcr) [inline lib/rules/jsx-no-target-blank.js]: @ljharb  Ho ok, my bad. No reason to not using it then.

Comment (randycoulman) [inline lib/rules/jsx-no-target-blank.js]: Do you want this as part of this PR, or should it be a separate one?  Feels unrelated to the bug IMO, but I'm willing to do it if you want.

Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: I'm fine with separate, but either way :-)

Comment (randycoulman) [inline lib/rules/jsx-no-target-blank.js]: I looked at this today.  There is a `break` inside the `for` loop which doesn't work in the `forEach` version.  I'm sure there's a way around that, but I think this should be in a separate PR.  I'd like to have this one considered done and merged (if acceptable).

Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: Yeah that'd require `.some` or similar. Separate PR is fine.

**Comment by yannickcr** [comment]
Merged, thanks!
